src/fsharp/FSharp.Core/reflect.fs

```diff
@@ -822,77 +736,86 @@ type UnionCaseInfo(typ: System.Type, tag:int) =
 type FSharpType = 
 
     static member IsTuple(typ:Type) =  
-        Impl.checkNonNull "typ" typ;
+        Impl.checkNonNull "typ" typ
         Impl.isTupleType typ
 
     static member IsRecord(typ:Type,?bindingFlags) =  
         let bindingFlags = defaultArg bindingFlags BindingFlags.Public 
 
-        Impl.checkNonNull "typ" typ;
+        Impl.checkNonNull "typ" typ
         Impl.isRecordType (typ,bindingFlags)
 
     static member IsUnion(typ:Type,?bindingFlags) =  
         let bindingFlags = defaultArg bindingFlags BindingFlags.Public 
-        Impl.checkNonNull "typ" typ;
+        Impl.checkNonNull "typ" typ
         let typ = Impl.getTypeOfReprType (typ ,BindingFlags.Public ||| BindingFlags.NonPublic)
         Impl.isUnionType (typ,bindingFlags)
 
     static member IsFunction(typ:Type) =  
-        Impl.checkNonNull "typ" typ;
+        Impl.checkNonNull "typ" typ
         let typ = Impl.getTypeOfReprType (typ ,BindingFlags.Public ||| BindingFlags.NonPublic)
         Impl.isFunctionType typ
 
     static member IsModule(typ:Type) =  
-        Impl.checkNonNull "typ" typ;
+        Impl.checkNonNull "typ" typ
         Impl.isModuleType typ
 
     static member MakeFunctionType(domain:Type,range:Type) = 
-        Impl.checkNonNull "domain" domain;
-        Impl.checkNonNull "range" range;
+        Impl.checkNonNull "domain" domain
+        Impl.checkNonNull "range" range
         Impl.func.MakeGenericType [| domain; range |]
 
     static member MakeTupleType(types:Type[]) =  
-        Impl.checkNonNull "types" types;
+        Impl.checkNonNull "types" types
+
+        // No assembly passed therefore just get framework local version of Tuple
+        let asm = typeof<System.Tuple>.Assembly
+        if types |> Array.exists (function null -> true | _ -> false) then 
+            invalidArg "types" (SR.GetString(SR.nullsNotAllowedInArray))
+        Impl.mkTupleType false asm types
+
+    static member MakeTupleTypeWithAssembly (asm:Assembly) (types:Type[]) =
```

Comment:
Please make this non-curried, i.e.

      static member MakeTupleTypeWithAssembly (asm:Assembly, types:Type[]) =

FSharp.Core doesn't currently use any curried members. Likewise MakeStructTupleType below.